doc/specs/#190 - Proxy Support.md

```diff
@@ -0,0 +1,86 @@
+---
+author: Flor Chacon @florelis
+created on: 2024-02-07
+last updated: 2024-02-07
+issue id: 190
+---
+
+# Proxy support
+
+For [#190](https://github.com/microsoft/winget-cli/issues/190)
+
+## Abstract
+
+This spec describes a feature to specify a proxy for winget to use when connecting to the internet.
+
+## Solution Design
+
+A new command line argument will be added to specify a proxy to use during a particular invocation of winget.
+
+An option to set a default proxy to use on every flow will be added, but it will be gated to require administrator permissions.
+
+New Group Policy will also be added for IT admins to control the use of proxies.
+The policies will be similar to those we already have for sources, so that a specific proxy can be required or only a predefined set of proxies can be allowed.
+
+Proxies will not be used for the configuration features for now.
```

Comment:
I think if we do the proxy configuration at wininet, DO, restclient level, then winget configuration should already be covered. Is there something I missed?

Reply:
I'm not familiar with the configuration code so I may be wrong, but the way I understand it is that that part of the code is mostly independent of the "package manager" side of things and some of it is written in .net. That's why I didn't include it in this.

I also don't know what network connections are done on that side. The only one that comes to mind is downloading the PS modules for each resource. If that's the only one, I don't see much case on going through the work of making it available on that side

Reply:
If PowerShell allows control over a proxy, then we could tunnel settings along to it.  Currently, the proxy would only affect the ability to retrieve a configuration document from the internet.

Reply:
I'm fine with extending the use of proxies to configuration, but it doesn't make much sense to me if it will only affect retrieving the configuration file.

The [`Install-Module`](https://learn.microsoft.com/powershell/module/powershellget/install-module) cmdlet has a `-Proxy`  argument but it "ignores this parameter since it's not supported by `Install-PSResource`," so it doesn't really help.

Apparently we could also set the [`WebRequest.DefaultWebProxy` property](https://learn.microsoft.com/en-us/dotnet/api/system.net.webrequest.defaultwebproxy) to set it for the whole PSSession, but we'd need to try it to be sure it works for this.

Reply:
btw, I'm fine with leaving configuration as future. I was just asking a naive question and I did not mean to add scope on this work.